packages/core-mobile/app/services/walletconnectv2/WalletConnectService.ts

```diff
@@ -246,22 +253,30 @@ class WalletConnectService implements WalletConnectServiceInterface {
     })
 
     // emitting events
+    // but only for evm chains since neither wagmi/universal provider can handle non-evm chain events
+    if (blockchainNamespace !== BlockchainNamespace.EIP155) {
+      Logger.info(
+        'skipping emitting wallet connect events since it is for a non-evm chain'
+      )
+      return
+    }
+
     await this.client.emitSessionEvent({
```

Comment:
unfortunately, still have to skip emitting events since from my testing, both wagmi and universal provider couldn't handle non-evm events.